examples/crm/src/dataProvider.ts

```diff
@@ -24,45 +24,60 @@ const TASK_MARKED_AS_UNDONE = 'TASK_MARKED_AS_UNDONE';
 const TASK_DONE_NOT_CHANGED = 'TASK_DONE_NOT_CHANGED';
 let taskUpdateType = TASK_DONE_NOT_CHANGED;
 
-const processLogo = async (params: any) => {
-    if (typeof params.data.logo !== 'object' || params.data.logo === null) {
-        return getCompanyAvatar(params.data);
-    }
-    const logo = params.data.logo;
-    if (logo.rawFile instanceof File) {
+const processCompanyLogo = async (params: any) => {
+    let logo = params.data.logo;
+
+    if (typeof logo !== 'object' || logo === null) {
+        logo = getCompanyAvatar(params.data);
+    } else if (logo.rawFile instanceof File) {
         const base64Logo = await convertFileToBase64(logo);
-        return {
-            src: base64Logo,
-            title: logo.title,
-        };
+        logo = { src: base64Logo, title: logo.title };
     }
 
-    return logo;
+    return {
+        ...params,
+        data: {
+            ...params.data,
+            logo,
+        },
+    };
 };
 
-const beforeContactUpsert = async (
-    params: CreateParams<any> | UpdateParams<any>,
+async function processContactAvatar(
+    params: UpdateParams<Contact>,
+    dataProvider: DataProvider
+): Promise<UpdateParams<Contact>>;
+
+async function processContactAvatar(
+    params: CreateParams<Contact>,
     dataProvider: DataProvider
-) => {
+): Promise<CreateParams<Contact>>;
+
+async function processContactAvatar(
```

Comment:
The name is misleading, as the function does more than avatar (it also sets the company name). I'd split it in two.